### PR TITLE
Redux recompose

### PR DESCRIPTION
**Pull request by alejobh**
## Summary
Added redux code refactory to Wolox's redux recompose code.
Changed some internal logic of components that needs a Loading state avoiding to re-render everything according to the React's reconcilier philosophy.
Changed some variables and functions names to a better approach.

## Trello Cards
https://trello.com/c/AGtQH4il

### CODE REVIEW

File: react/game/src/app/screens/Game/index.js [hunk 1]
```diff
@@ -11,10 +11,15 @@ import GameLayout from './layout';
...
   componentDidMount() {
     this.props.loadGameInfo();
-    store.subscribe(() => {
+    this.unsuscribe = store.subscribe(() => {
       saveGameState(store.getState().game);
     });
   }
+
+  componentWillUnmount() {
+    this.unsuscribe();
```
Comment (wfolini) [inline react/game/src/app/screens/Game/index.js:20]: 👌

File: react/game/src/redux/auth/actions.js [hunk 2]
```diff
@@ -1,76 +1,52 @@
...
-};
+const completedTypes = completeTypes(['INIT_APP_LOADING', 'LOGIN', 'LOGOUT'], ['INIT_APP_LOADING']);
+
+export const actions = createTypes(completedTypes, '@@AUTH');
 
 export const actionCreators = {
-  initApp: () => async dispatch => {
-    dispatch(privateActionCreators.appLoaded(false));
+  initApp: () => dispatch => {
     const session = loadAuthState();
```
Comment (wfolini) [inline react/game/src/redux/auth/actions.js:16]: U need handle async all local storage services

File: react/game/src/redux/game/selectors.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { concatHistory, calculateWinner } from '@utils/game';
```
Comment (wfolini) [inline react/game/src/redux/game/selectors.js]: I'd rename this file `effects`. Selectors are something else that does enter in the scope of this card, it is other approach in an optional trello card

File: react/game/src/redux/profile/actions.js [hunk 4]
```diff
@@ -1,58 +1,41 @@
...
-      dispatch(privateActionCreators.error(data.error.message));
-      throw new SubmissionError({ _error: data.error.message });
-    }
-  }
+  loadProfile: id => ({
+    type: actions.PROFILE_INFO,
+    service: profileService.get,
+    payload: id,
+    target: TARGET,
+    successSelector: response => response.data,
```
Comment (wfolini) [inline react/game/src/redux/profile/actions.js]: Delete this selector. It's the default selector

File: react/game/src/redux/profile/actions.js [hunk 5]
```diff
@@ -1,58 +1,41 @@
...
+          type: actions.SET_SUCCESS_MESSAGE,
+          target: TARGET_SUCCESS_MESSAGE,
+          payload: formNames.EDIT_PROFILE.messages.success
+        });
+      }),
+      withPostFailure((dispatch, response) => {
+        throw new SubmissionError({ _error: response.data.error.message });
+      })
+    ],
+    successSelector: response => response.data,
```
Comment (wfolini) [inline react/game/src/redux/profile/actions.js]: Also this

**Review by wfolini — CHANGES_REQUESTED** [review]
Excellent! 👏! Only few comments

File: react/game/src/redux/auth/actions.js (hunk 2 above)
Comment (alejobh) [reply react/game/src/redux/auth/actions.js:16]: According to the HML5 API Docs, all the local storage calls are asynchronous, read this https://stackoverflow.com/questions/20231163/is-html5-localstorage-asynchronous

File: react/game/src/redux/profile/actions.js (hunk 5 above)
Comment (alejobh) [reply react/game/src/redux/profile/actions.js]: Thanks, i did not know it! 👌🏼

**Review by wfolini — DISMISSED** [review]

File: react/game/src/redux/auth/actions.js [hunk 6]
```diff
@@ -1,76 +1,52 @@
...
 
-export const AUTH_ACTIONS = {
-  LOAD_APP: 'LOAD_APP',
-  APP_LOADED: 'APP_LOADED',
-  LOGIN_LOADING: 'LOGIN_LOADING',
-  LOGIN_ERROR: 'LOGIN_ERROR',
-  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
-  LOGOUT: 'LOGOUT'
-};
+import { TARGET_APP_LOADING, TARGET } from './constants';
```
Comment (jmtov) [inline react/game/src/redux/auth/actions.js]: I like this idea! However I'd just export an object from the constant files named `targets` with keys named as the target names, so it'd be easier to understand. 😺

File: react/game/src/redux/profile/actions.js [hunk 7]
```diff
@@ -1,58 +1,39 @@
...
 import { SubmissionError } from 'redux-form';
 import formNames from '@constants/formNames';
 
-export const PROFILE_ACTIONS = {
-  SET_PROFILE_INFO: 'SET_PROFILE_INFO',
-  LOADING: 'LOADING',
-  ERROR: 'ERROR',
-  SUCCESS: 'SUCCESS'
-};
+import { TARGET, TARGET_SUCCESS_MESSAGE } from './constants';
```
Comment (jmtov) [inline react/game/src/redux/profile/actions.js]: Same here regarding the `target` comment.

**Review by jmtov — CHANGES_REQUESTED** [review]
Awesome job, Alejo! I just have a comment, please check it out! :)

File: react/game/src/redux/auth/actions.js (hunk 6 above)
Comment (alejobh) [reply react/game/src/redux/auth/actions.js]: Thanks for your comment!

File: react/game/src/redux/profile/actions.js (hunk 7 above)
Comment (alejobh) [reply react/game/src/redux/profile/actions.js]: 👌🏼

**Review by wfolini — DISMISSED** [review]

File: react/game/src/redux/auth/constants.js [hunk 8]
```diff
@@ -1,2 +1,6 @@
-export const TARGET_APP_LOADING = 'appLoading';
-export const TARGET = 'authInfo';
+const targets = {
+  TARGET: 'authInfo',
```
Comment (jmtov) [inline react/game/src/redux/auth/constants.js]: This is better! But keep in mind that it'd be easier to understand your code if the names you use are a bit more descriptive. :)

**Review by jmtov — CHANGES_REQUESTED** [review]

File: react/game/src/redux/auth/constants.js (hunk 8 above)
Comment (alejobh) [reply react/game/src/redux/auth/constants.js]: solved, sorry i had not understood the changes that you had requested.

**Review by jmtov — APPROVED** [review]
Awesome! 😺🤙

**Pull request merged**